sparkler-app/src/main/scala/edu/usc/irds/sparkler/pipeline/Crawler.scala

```diff
@@ -130,55 +135,101 @@ class Crawler extends CliTool {
   //TODO: Robots.txt
 
   override def run(): Unit = {
-
     //STEP : Initialize environment
     init()
 
     val solrc = this.job.newCrawlDbSolrClient()
+    LOG.info("Committing crawldb..")
+    solrc.commitCrawlDb()
     val localFetchDelay = fetchDelay
     val job = this.job // local variable to bypass serialization
 
     for (_ <- 1 to iterations) {
-      val taskId = JobUtil.newSegmentId(true)
+      var taskId = JobUtil.newSegmentId(true)
       job.currentTask = taskId
       LOG.info(s"Starting the job:$jobId, task:$taskId")
 
-
       val rdd = new CrawlDbRDD(sc, job, sortBy = sortBy, maxGroups = topG,
         topN = topN, groupBy = groupBy)
       val fetchedRdd = rdd.map(r => (r.getGroup, r))
         .groupByKey()
         .flatMap({ case (grp, rs) => new FairFetcher(job, rs.iterator, localFetchDelay,
-          FetchFunction, ParseFunction, OutLinkFilterFunction) })
+          FetchFunction, ParseFunction, OutLinkFilterFunction)
+        })
         .persist()
 
       if (kafkaEnable) {
         storeContentKafka(kafkaListeners, kafkaTopic.format(jobId), fetchedRdd)
       }
 
-      //Step: Update status of fetched resources
+      //Step: Update status of fetched resources.
       val statusUpdateRdd: RDD[SolrInputDocument] = fetchedRdd.map(d => StatusUpdateSolrTransformer(d))
       val statusUpdateFunc = new SolrStatusUpdate(job)
       sc.runJob(statusUpdateRdd, statusUpdateFunc)
 
-      //Step: Filter Outlinks and Upsert new URLs into CrawlDb
-      val outlinksRdd = OutLinkUpsert(job, fetchedRdd)
-      val upsertFunc = new SolrUpsert(job)
-      sc.runJob(outlinksRdd, upsertFunc)
-
+      val scoredRdd = score(fetchedRdd)
       //Step: Store these to nutch segments
       val outputPath = this.outputPath + "/" + taskId
 
-      storeContent(outputPath, fetchedRdd)
+      storeContent(outputPath, scoredRdd)
 
       LOG.info("Committing crawldb..")
       solrc.commitCrawlDb()
     }
+
     solrc.close()
     //PluginService.shutdown(job)
     LOG.info("Shutting down Spark CTX..")
     sc.stop()
   }
+
+  def executePipeline(rdd: RDD[Resource], taskId: String, solrc: SolrProxy, localFetchDelay: Long): Unit = {
```

Comment:
where is the `executePipeline` actually called? Is it really needed?

Reply:
I think this has been removed, dont know if it is committed. @giuseppetotaro ? 
It was created since we were trying to add more code in the run method which resulted in 50+ lines, so split out the actual crawl part.